paystand/paystand-magento2
# PPL-193: Magento - Create Transaction when a payment is done

JIRA
----
https://paystand.atlassian.net/browse/PPL-193

Description
----
This PR add support for creating "capture transactions" and "invoices" when an order is paid with paystand.
**Details**
- Capture Transaction is created, it contains the paystand payment information: transactionId, amount, currency, paymentStatus
- Invoice is created for the order automatically after Magento store receives the "paid" status webhook for the payment
- Notes for the order are now created:
![image](https://user-images.githubusercontent.com/33076382/88204240-9c2d9f00-cc10-11ea-9bfa-b7abdbc9c195.png)
- Order amounts are now updated: amount due goes to 0 after the payment is done, paid amount is now updated too 
![image](https://user-images.githubusercontent.com/33076382/88204327-b8c9d700-cc10-11ea-83b5-5c240afea19a.png)

QA Notes
----
**Check with Jose before QA to make sure our magento instance "magento.paystand.biz" has this code.
**GENERAL INSTRUCTIONS**
1) Go to magento.paystand.biz
2) Add a product to your cart, and follow the process of buying it paying with paystand
3) after you completed your purchase and you get to the "**Thank you page**" take note of the order number and then login to http://magento.paystand.biz/admin_1i8kyl (credentials: user: admin, password: admin_1i8kyl)
4) Go to "**Sales**" -> "**Orders**" page, and click on the order you just created
5) Verify the following information
- The order should be on "Processing" status (it may take a minute or two after you made the purchase)
- At the very end of the page, you should see Notes for the order, you should see something like this:
![image](https://user-images.githubusercontent.com/33076382/88204240-9c2d9f00-cc10-11ea-9bfa-b7abdbc9c195.png)
- On order Totals, you should see that the "**Total Due**" is 0, and that "**Total Paid**" is the same as "**Grand Total**"
![image](https://user-images.githubusercontent.com/33076382/88204327-b8c9d700-cc10-11ea-83b5-5c240afea19a.png)
- if you click on "**Transactions**" tab you should see a "**Capture**" transaction, if you click on it, you will get more details, under Transaction Details you should see the following information:
![image](https://user-images.githubusercontent.com/33076382/88204444-e151d100-cc10-11ea-917d-6f2200b9dc15.png)
- Go back to the order, and now click on "**Invoices**", you should see a "**Paid**" invoice created in there
